src/Microsoft.AspNetCore.Mvc.Core/StatusCodeResult.cs

```diff
@@ -13,7 +13,7 @@ namespace Microsoft.AspNetCore.Mvc
     /// Represents an <see cref="ActionResult"/> that when executed will
     /// produce an HTTP response with the given response status code.
     /// </summary>
-    public class StatusCodeResult : ActionResult, IClientErrorActionResult
+    public class StatusCodeResult : ActionResult, IStatusCodeActionResult
```

Comment:
This was weird - not all status code results are client errors. Moving the interface to the 4xx ones